File: tasks/core-java-quota/1.0.0/workspace/Quota.java
public final class Quota {
    private int remaining;

    public Quota(int capacity) { this.remaining = capacity; }
    public int remaining() { return remaining; }

    public boolean consume(int units) {
        if (units < remaining) {
            remaining -= units;
            return true;
        }
        remaining -= units;
        return false;
    }
}

File: tasks/core-python-quota/1.0.0/oracle/quota.py
class Quota:
    def __init__(self, capacity: int) -> None:
        self.remaining = capacity

    def consume(self, units: int) -> bool:
        if units <= 0:
            raise ValueError("units must be positive")
        if units > self.remaining:
            return False
        self.remaining -= units
        return True
